pkg/healthcheck/healthcheck.go

```diff
@@ -988,6 +1004,43 @@ func (hc *HealthChecker) checkNetAdmin() error {
 	return fmt.Errorf("found %d PodSecurityPolicies, but none provide NET_ADMIN", len(pspList.Items))
 }
 
+func (hc *HealthChecker) checkClockSkew() error {
+	if hc.kubeAPI == nil {
+		// we should never get here
+		return fmt.Errorf("unexpected error: Kubernetes ClientSet not initialized")
+	}
+
+	var clockSkewNodes []string
+
+	nodeList, err := hc.kubeAPI.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, node := range nodeList.Items {
+		for _, condition := range node.Status.Conditions {
+			// we want to check only KubeletReady condition and only execute if the node is ready
+			if condition.Type == "Ready" && condition.Status == "True" {
+				ts := condition.LastHeartbeatTime.Time
+				if (time.Since(ts) > AllowedClockSkew) || (time.Since(ts) < -AllowedClockSkew) {
```

Comment:
tioli:
```go
since := time.Since(condition.LastHeartbeatTime.Time)
if (since > AllowedClockSkew) || (since < -AllowedClockSkew) {
```
(also i can't think of an duration-type-safe-way to validate magnitude with a single comparison :p )